src/convertToObject.js

```diff
@@ -6,7 +6,26 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const styleObject = {};
+  const cleanStrings = sourceString
+    .trim()
+    .split(';')
+    .filter((style) => style.length);
+
+  cleanStrings.forEach((cleanString) => {
+    const partsOfString = cleanString
+      .split(':')
+      .map((partStyle) => partStyle.trim());
+
+    const key = partsOfString[0];
+    const value = partsOfString[1];
+
+    if (key && value) {
+      styleObject[key] = value;
+    }
+  });
+
+  return styleObject;
```

Comment:
u can do it easier using `reduce` method
u will avoid using 2 extra iterations of `filter ` and `map `
write all the logic in the middle of the `reduce' method

```suggestion
  return sourceString.split(';').reduce((style, partsOfString) => {
  
    ..............

    return style;
  }, {});
```